components/core/src/clp_s/JsonParser.hpp

```diff
@@ -38,6 +53,14 @@ struct JsonParserOption {
     std::shared_ptr<clp::GlobalMySQLMetadataDB> metadata_db;
 };
 
+struct JsonToIRParserOption {
+    std::vector<std::string> file_paths;
+    std::string irs_dir;
+    size_t max_document_size;
+    int compression_level;
+    int encoding;
+};
```

Comment:
This could be good to consider for later, but I am following the precedent set  in the JsonParserOption, but we could consider this change for both structs later.